libyul/optimiser/SSATransform.cpp

```diff
@@ -40,11 +40,7 @@ void SSATransform::operator()(Identifier& _identifier)
 
 void SSATransform::operator()(ForLoop& _for)
 {
-	// This will clear the current value in case of a reassignment inside the
```

Comment:
I think it is better to keep the SSATransform handle the pre loop - it is not a complication for it, just an inefficiency. Also ForLoopInitRewriter should not be a requirement for the SSA transform.